sdk/src/Core/Amazon.Runtime/Pipeline/RetryHandler/RetryPolicy.cs

```diff
@@ -446,27 +445,28 @@ private bool IsClockskew(IExecutionContext executionContext, Exception exception
                     // using accurate server time, calculate correction if local time is off
                     serverTime = serverTime.ToUniversalTime();
                     var diff = correctedNow - serverTime;
+                    var newCorrection = serverTime - realNow;
                     var absDiff = diff.Ticks < 0 ? -diff : diff;
-                    if (absDiff > clockSkewMaxThreshold)
-                    {
-                        var newCorrection = serverTime - realNow;
-                        Logger.InfoFormat(clockSkewMessageFormat,
-                            realNow, correctedNow, clientConfig.ClockOffset, serverTime, endpoint);
-
-                        // Always set the correction, for informational purposes
-                        CorrectClockSkew.SetClockCorrectionForEndpoint(endpoint, newCorrection);
 
-                        var shouldRetry = AWSConfigs.CorrectForClockSkew && !AWSConfigs.ManualClockCorrection.HasValue;
+                    Logger.InfoFormat(clockSkewMessageFormat,
+                        realNow, correctedNow, clientConfig.ClockOffset, serverTime, endpoint);
 
-                        // Only retry if clock skew correction is not disabled
-                        if (shouldRetry)
+                    // Always set the correction, for informational purposes
+                    CorrectClockSkew.SetClockCorrectionForEndpoint(endpoint, newCorrection);
+                    var shouldRetry = AWSConfigs.CorrectForClockSkew && !AWSConfigs.ManualClockCorrection.HasValue;
+                    // Some services such as S3 will return a generic 403 error for clock skew errors on HEAD requests
+                    // with no error code. In this case we will always retry the request if the clock skew is greater than
+                    // the threshold.
+                    if (shouldRetry)
+                    {
+                        if ((isHead && absDiff > clockSkewMaxThreshold) || isClockskewErrorCode)
```

Comment:
As per this code, the bug is not fixed for S3. In the repro scenario when 2 threads are coming together, the second one will not go inside the if block and will return false causing it to not retry and the throw the exception back to caller of SDK

Reply:
ack, let's unblock you guys first and come back to the head request as a follow up.